FJAM-Studios/pixtery
# Run once function

This PR is related to the relative image path PR and is meant to be merged after that one. 

The goal of this PR is to run a function once that:
- moves Pixtery images stored on your device into the appropriate folder (documentDirectory instead of cacheDirectory)
- renames those images to have a jpg extension (instead of no extension)
- updates the puzzle data to point to the right place (the relative path instead of absolute)

It's a bit tricky to test, but here's how I've done it:
- first clear your sent and received puzzles
- git checkout main (not dev, since main is the one that users are on "in the wild")
- npm start
- make some Pixteries, and send them to yourself. Now you have Pixteries stored in the wrong place and with no file extension, just like our users.
- close Pixtery and kill the expo server
- git checkout image-relative-path
- npm start (may have to npm install, if you don't have everything)
- note that the app still shows the puzzles you created on main but can't find any of the images because they have the wrong paths
- close Pixtery and kill the expo server
- git checkout run-once-function
- npm start and note the console logs. They should indicate that the function is being run and moving/renaming those images
- now you've got your puzzles from before AND you can save the images to your gallery, even if they were made pre-update!

I've only tested this on Android, but, fingers crossed, it works the same on iOS and in production. My hope is that this can be pushed to the expo channel and my Android app which is currently on 1.0.5 but has Pixteries from 1.0.3 will once again have access to those Pixteries. Of course it's possible that the Android update already wiped out the old Pixtery directory, but at least this will move everyone's images to the right directory and give them an extension so that they can be saved to the gallery.
